MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -4628,66 +4619,44 @@ ApplicationWindow *ApplicationWindow::openProject(const QString &filename,
 
   d_opening_file = true;
 
-  QFile file(filename);
-  QFileInfo fileInfo(filename);
-
-  if (!file.open(QIODevice::ReadOnly))
-    throw std::runtime_error("Couldn't open project file");
-
-  QTextStream fileTS(&file);
-  fileTS.setCodec(QTextCodec::codecForName("UTF-8"));
-
-  QString baseName = fileInfo.fileName();
-
-  // Skip mantid version line
-  fileTS.readLine();
-
-  // Skip the <scripting-lang> line. We only really use python now anyway.
-  fileTS.readLine();
-  setScriptingLanguage("Python");
-
-  // Skip the <windows> line.
-  fileTS.readLine();
-
   folders->blockSignals(true);
   blockSignals(true);
 
+  // Open as a top level folder
+  ProjectSerialiser serialiser(this);
+  try {
+    serialiser.load(filename.toStdString(), fileVersion);
+  } catch (std::runtime_error &e) {
```

Comment:
Is this guaranteed to only throw a `runtime_error`?

Reply:
It's the only exception type that we throw from `ProjectSerialiser`. 
Anything else is probably a hard error (such as OOM) we probably don't want to attempt any recovery from

Reply:
:+1: